contracts/FastBreak.cdc

```diff
@@ -0,0 +1,802 @@
+        ///
+        access(contract) fun updateScore(playerId: UInt64, points: UInt64, win: Bool): Bool {
+            let submissions = self.submissions
+
+            let submission: FastBreak.FastBreakSubmission = submissions[playerId]
+                ?? panic("unable to find fast break submission for playerId")
+
+            submission.setPoints(points: points, win: win)
+
+            self.submissions[playerId] = submission
+
+            if win && !submission.win {
+                return true
+            }
+
+            return false
+        }
+    }
+
+    /// Validate Fast Break Submission
+    ///
+    pub fun isValidSubmission(submissionDeadline: UInt64): Bool {
+        return submissionDeadline > UInt64(getCurrentBlock().timestamp) 
+    }
+
+    /// Get a Fast Break Game by Id
+    ///
+    pub fun getFastBreakGame(id: String): FastBreak.FastBreakGame? {
+        return FastBreak.fastBreakGameByID[id]
+    }
+
+    /// Get the game stats of a Fast Break
+    ///
+    pub fun getFastBreakGameStats(id: String): [FastBreak.FastBreakStat] {
+        if let fastBreak = FastBreak.getFastBreakGame(id: id) {
+            return fastBreak.stats
+        }
+        return []
+    }
+
+    /// A statistical structure used in Fast Break Games
+    /// This structure names the NBA statistic top shots must match or exceed
+    /// An example is points as the statistic and 30 as the value
+    /// A top shot or group of top shots must meet or exceed 30 points
+    ///
+    pub struct FastBreakStat {
+        pub let name: String
+        pub let type: FastBreak.StatisticType
+        pub let valueNeeded: UInt64
+
+        init (
+            name: String,
+            type: FastBreak.StatisticType,
+            valueNeeded: UInt64
+        ) {
+            self.name = name
+            self.type = type
+            self.valueNeeded = valueNeeded
+        }
+    }
+
+    /// An account submission to a Fast Break
+    ///
+    pub struct FastBreakSubmission {
+        pub let playerId: UInt64
+        pub var submittedAt: UInt64
+        pub let fastBreakGameID: String
+        pub var topShots: [UInt64]
+        pub var points: UInt64
+        pub var win: Bool
+
+        init (
+            playerId: UInt64,
+            fastBreakGameID: String,
+            topShots: [UInt64],
+        ) {
+            self.playerId = playerId
+            self.fastBreakGameID = fastBreakGameID
+            self.topShots = topShots
+            self.submittedAt = UInt64(getCurrentBlock().timestamp)
+            self.points = 0
+            self.win = false
+        }
+
+        /// Set the points of a submission
+        ///
+        access(contract) fun setPoints(points: UInt64, win: Bool) {
+            self.points = points
+            self.win = win
+        }
+
+    }
+
+    /// Resource for playing Fast Break
+    /// The Fast Break Player plays the game & mints game tokens
+    ///
+    pub resource Player: FastBreakPlayer {
```

Comment:
```suggestion
    pub resource Player: FastBreakPlayer, NonFungibleToken.INFT {
```
I think it makes sense to make this an NFT also. No need to have a collection defined for it too, though, as long as it implements the interface, that should be fine, especially once Cadence 1.0 comes around